src/main/java/net/minecraftforge/common/ForgeMod.java

```diff
@@ -181,13 +193,139 @@ public void gatherData(GatherDataEvent event)
     {
         DataGenerator gen = event.getGenerator();
 
+        if (event.includeClient())
+        {
+            gen.addProvider(new ItemModels(gen, event.getExistingFileHelper()));
+            gen.addProvider(new BlockStates(gen, event.getExistingFileHelper()));
+        }
         if (event.includeServer())
         {
             gen.addProvider(new ForgeBlockTagsProvider(gen));
             gen.addProvider(new ForgeItemTagsProvider(gen));
             gen.addProvider(new ForgeRecipeProvider(gen));
         }
     }
+    
+    public static class ItemModels extends ModelProvider<ItemModelBuilder>
```

Comment:
All of this should go to a test mod

Reply:
As noted in the OP, it's temporary because the forge_test_data run config does not seem to work at the moment.

Reply:
they do in my tests... You should look into that.